web-common/src/features/sources/modal/AddDataForm.svelte

```diff
@@ -150,6 +156,46 @@
   // Emit the submitting state to the parent
   $: dispatch("submitting", { submitting });
 
+  // Generate YAML preview from form state
+  $: yamlPreview = (() => {
+    if (connector.name === "clickhouse") {
```

Comment:
It's not immediately obvious to me why the logic would be different for ClickHouse than it would be for other connectors. Could you add a comment explaining why? Is this a short-term thing?

Reply:
I understand your concern. The decision to separate Clickhouse from Data Form was made to address the distinct responsibilities of their respective UI rendering processes. This change was primarily driven by the need to accommodate the conditional rendering of the "managed" field.﻿

Reply:
Okay, again sounds like a JSON Schema might help us out here